### DISCUSSION TITLE
@pothos/plugin-prisma with typescript as devDependency

### DISCUSSION THREAD

**Discussion by nephix** [hayes/pothos#1075] [Q&A] [answered]
Hi 👋 

I'm using pothos and the `@pothos/plugin-prisma` plugin to generate types from prisma schema. This works well in development as `typescript` is listed under `devDependencies`:

```json
"devDependencies": {
    "typescript": "^5.1.3",
    ...
  },
  "dependencies": {
    "@pothos/core": "^3.38.0",
    "@pothos/plugin-prisma": "^3.61.0",
    "@prisma/client": "^5.5.2",
    "prisma": "^5.5.2",
    ...
  },
```

but when building our backend docker image for production I run into an issue: I need to run

```
npx prisma generate
```

to generate the prisma client within `node_modules`.

But since we use the pothos prisma plugin, our `schema.prisma` includes:

```
generator pothos {
  provider     = "prisma-pothos-types"
  clientOutput = "@prisma/client"
  output       = "../src/generated/types.ts"
}
```

which requires `typescript` to be listed under `dependencies` instead of `devDependencies` (we omit installation of `devDependencies` via `npm install --omit=dev`)

I know that we can generate the prisma client into a different location so that it's part of the repository, [but that seems to have it's own drawbacks](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#why-is-prisma-client-generated-into-node_modulesprismaclient-by-default). 

Is there a way to skip the generation of the `prisma-pothos-types` provider depending on an env var like `NODE_ENV`? Or any better solutions?

Cheers

**Comment by nephix** [comment, marked as answer]
Ok it doesn't seem to be possible without saving the generated client outside of `node_modules`

**Reply by hayes** [reply]
This isn't currently a feature, but happy to accept a PR to add it

**Reply by nephix** [reply]
@hayes I'd be willing to do this, but I may need a couple of pointers into right direction. What's the best way to get in touch with you?

**Reply by hayes** [reply]
The discord channel (linked fr.the docs site) works well.  I'm in the middle of moving right now, so my availability is a bit scattered

**Reply by nephix** [reply]
I don't really use Discord. Would you like me to create an issue for this instead

**Reply by hayes** [reply]
Yeah, GitHub issue is fine too!
